devimint/src/gatewayd.rs

```diff
@@ -380,26 +386,37 @@ impl Gatewayd {
         Ok(channels)
     }
 
-    pub async fn wait_for_chain_sync(&self, bitcoind: &Bitcoind) -> Result<()> {
-        poll("lightning node block processing", || async {
-            let block_height = bitcoind
-                .get_block_count()
-                .await
-                .map_err(ControlFlow::Continue)?
-                - 1;
-            cmd!(
-                self,
-                "lightning",
-                "wait-for-chain-sync",
-                "--block-height",
-                block_height
-            )
-            .run()
-            .await
-            .map_err(ControlFlow::Continue)?;
+    pub async fn wait_for_chain_sync(&self) -> Result<()> {
+        poll("waiting for gatewayd chain sync", || async {
+            let info = self.get_info().await.map_err(ControlFlow::Continue)?;
+            let gateway_info: GatewayInfo =
+                serde_json::from_value(info).expect("Failed to decode GatewayInfo");
+            if gateway_info.gateway_state != "Running" || !gateway_info.synced_to_chain {
+                return Err(ControlFlow::Continue(anyhow::anyhow!(
+                    "Not synced to chain"
+                )));
+            }
             Ok(())
         })
         .await?;
         Ok(())
     }
+
+    pub async fn wait_for_block_height(&self, target_block_height: u64) -> Result<()> {
+        poll("waiting for block height", || async {
+            let info = self.get_info().await.map_err(ControlFlow::Continue)?;
+            let gateway_info: GatewayInfo =
+                serde_json::from_value(info).expect("Failed to decode GatewayInfo");
+            if let Some(height) = gateway_info.block_height {
+                if height >= target_block_height as u32 && gateway_info.synced_to_chain {
```

Comment:
Is `synced_to_chain` strictly necessary? Is the `height` reported the synced height or the height that the node is aware of but not necessarily synced to yet? If it's the former then just syncing to the height should be sufficient.

In cases where parallel processes keep mining blocks the "synced to chain" state might not be reached for quite a while.

Reply:
Agreed, I don't think it is necessary